syntax = "proto3";

package routesia.route;

// Next hop gateway
//
message RouteNextHop {
    // Gateway
    //
    string gateway = 1;

    // Interface
    //
    string interface = 2;

    // Hops (For multipath)
    //
    uint32 hops = 3;
}

message RouteState {
    // Is present
    //
    bool present = 1;

    // Table
    //
    uint32 table_id = 2;

    // Destination
    //
    string destination = 3;

    // Protocol
    //
    uint32 protocol = 4;

    // Scope
    //
    uint32 scope = 5;

    // Preferred source
    //
    string preferred_source = 6;

    // Next hops (gateways)
    //
    repeated routesia.route.RouteNextHop nexthop = 7;
}

// Static route configuration
//
message RouteConfig {
    // Destination
    //
    string destination = 1;

    // Next hops (gateways)
    //
    repeated routesia.route.RouteNextHop nexthop = 2;
}

message RouteTableConfig {
    // Name
    //
    string name = 1;

    // ID
    //
    uint32 id = 2;

    // Static routes
    //
    repeated RouteConfig static = 3;
}

// Route module config
//
message RouteModuleConfig {
    repeated RouteTableConfig table = 1;
}
